vdr/vdr.go

```diff
@@ -118,6 +122,17 @@ func (r *VDR) ConflictedDocuments() ([]did.Document, []types.DocumentMetadata, e
 	return conflictedDocs, conflictedMeta, err
 }
 
+func (r *VDR) IsOwner(ctx context.Context, id did.DID) (bool, error) {
+	// First perform a cheap DID existence check (subsequent checks are more expensive)
+	_, _, err := r.store.Resolve(id, nil)
```

Comment:
although cheap, it's not cached. My guess is that you can skip this step and make the average performance faster. The only reason to do this, is if you expect a lot of invalid requests. Also the Resolve will be probably be done again later in the flow.

I think it would be beneficial if the Resolve operation has it's own internal cache? >> new issue?

Reply:
Problem is not checking DID existence would introduce a DoS vulnerability for the negative result caching, since an attacker can just post random `did:nuts` strings, growing the negative result cache indefinitely.

When the store is in BBolt, it's already in-memory thus won't get much faster then it already is? (Maybe less locking).

Reply:
Should this move to `cachingDocumentOwner`? You probably want this check for every type of cached document owner.

Reply:
BBolt does not keep everything in cache, also this isn't true for other storage. That's why I suggested it to become a new issue.

Reply:
That's right. But i.m.o. it was still worth adding, since in the recommended production deployment (VDR storage in Redis, private keys in Vault through the Vault Proxy) querying the VDR storage is most probably faster than querying Vault.